MotionPhotoReader/app/src/main/java/com/example/motionphotoreader/XmpParser.java

```diff
@@ -0,0 +1,82 @@
+package com.example.motionphotoreader;
+
+import android.util.Log;
+
+import com.adobe.internal.xmp.XMPException;
+import com.adobe.internal.xmp.XMPMeta;
+import com.adobe.internal.xmp.XMPMetaFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+
+/**
+ * The XmpParser class is a package-private class intended to help extract the microvideo offset
+ * information from the XMP metadata of the given Motion Photo. This class is for internal use only.
+ */
+class XmpParser {
+
+    private static final byte[] OPEN_ARR = "<x:xmpmeta".getBytes();  /* Start of XMP metadata tag */
+    private static final byte[] CLOSE_ARR = "</x:xmpmeta>".getBytes();  /* End of XMP metadata tag */
+
+    /**
+     * Copies the input stream from a file to an output stream.
+     */
+    private static void copy(String filename, InputStream in, OutputStream out) throws IOException {
+        int len;
+        byte[] buf = new byte[1024];
+        while((len = in.read(buf)) >= 0) {
+            out.write(buf, 0, len);
+        }
+        in.close();
+        out.close();
```

Comment:
Ownership of InputStream and OutputStream in most cases should best be handled by the caller. That way, there isn't any ambiguity whether the arguments got closed. The exception here is if the method name has the phrase "close" or "release" in it (e.g. if this was copyAndClose).

So, avoid calling in.close and out.close in here.

Reply:
Deleted this method (using Guava Bytestreams.copy() instead)